examples/specs/layer_bar_annotations.vl.json

```diff
@@ -0,0 +1,60 @@
+{
+  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
+  "description": "A simple bar chart with annotation.",
+  "data": {
+    "values": [
+      {"Day": "D01", "Value": "50"},
+      {"Day": "D02", "Value": "110"},
+      {"Day": "D03", "Value": "70"},
+      {"Day": "D04", "Value": "50"},
+      {"Day": "D05", "Value": "80"},
+      {"Day": "D06", "Value": "150"},
+      {"Day": "D07", "Value": "140"},
+      {"Day": "D08", "Value": "130"},
+      {"Day": "D09", "Value": "390"},
+      {"Day": "D10", "Value": "200"},
+      {"Day": "D11", "Value": "70"},
+      {"Day": "D12", "Value": "50"},
+      {"Day": "D13", "Value": "100"},
+      {"Day": "D14", "Value": "150"},
+      {"Day": "D15", "Value": "130"},
+      {"Threshold": "hazardous", "ThresholdValue": "300"}
+    ]
+  },
+  "transform": [{
+    "calculate": "datum.Value > 300 ? 300 : datum.Value", "as": "diff"
```

Comment:
Why is this called "diff"?  The formula doesn't read like diff to me.  
Instead of calculating a new field like this, I think your red bar layer can just "filter" if value is more than 300 and encode data from y=300 to y=field: value.